cmd/idl-repository/cmd/root.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/syncromatics/idl-repository/internal/repository"
+	"github.com/syncromatics/idl-repository/internal/storage"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	port            *int
+	storageDiretory *string
+)
+
+func init() {
+	port = rootCmd.Flags().IntP("port", "p", 80, "The port to host the server on")
+	storageDiretory = rootCmd.Flags().StringP("storage", "s", "/tmp/storage", "The storage location for modules")
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "idl-repository",
+	Short: "idl-repository stores all sorts of idls",
+	Long:  `long explanation here`,
+	Run: func(cmd *cobra.Command, args []string) {
+		settings := &repository.Settings{
+			Port: *port,
+		}
+
+		storage, err := storage.NewFileStorage(*storageDiretory)
+		if err != nil {
+			panic(err)
+		}
+		server := repository.NewServer(settings, storage)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		grp, ctx := errgroup.WithContext(ctx)
+
+		grp.Go(server.Run())
+
+		// Wait for SIGINT/SIGTERM
+		waiter := make(chan os.Signal, 1)
+		signal.Notify(waiter, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
This doesn't seem to be respecting Ctrl+C.

Reply:
The web server isn't setup right to cancel when the context is done. I'll fix that

Reply:
This is fixed  by the changes here https://github.com/syncromatics/idl-repository/pull/1/files#diff-e16bea04351e58ac3fd0ab77080cd53dR70